gridt/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from "@angular/core";
+import { Observable, pipe, throwError, of } from "rxjs";
+import { flatMap, tap, map, catchError, pluck } from "rxjs/operators";
+
+import { Identity } from './models/identity.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class SettingsService {
+
+  constructor (
+    private auth: AuthService,
+    private secStore: SecureStorageService,
+  ) {}
+
+  /**
+   * Observable to obtain settings from secure storage
+   * If secure storage is empty, this returns an error
+   */
+  public localIdentity$: Observable<Identity> = this.auth.readyAuthentication$.pipe(
+    flatMap(() => this.secStore.get$("identity"))
+  );
+
+  /**
+   * Store settings in the secure storage.
+   * @param identity Settings to be stored.
+   */
+  public setLocalIdentity$(identity: Identity): Observable<boolean> {
+    console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
+    let loggedIn: boolean;
+    this.auth.isLoggedIn$.subscribe( (val) => {loggedIn = val;});
+
+    if (loggedIn) {
```

Comment:
This should always fail when you are online. Your test should make the wait time a little longer so that you can see that in your tests, but the subscription will only be called once the request responds. That might be a few cycles later, meaning that this will be evaluated to false immediately. The logic you have placed here should be in the subscription function.